src/internal/storage/chunk/transform.go

```diff
@@ -190,7 +190,7 @@ func cryptoXOR(key, dst, src []byte) {
 func verifyData(id ID, x []byte) error {
 	actualHash := Hash(x)
 	if !bytes.Equal(actualHash[:], id) {
-		return errors.Errorf("bad chunk. HAVE: %x WANT: %x", actualHash, id)
```

Comment:
This is the most important change in here.  This will prevent the double hex encoding.